### DISCUSSION TITLE
Is crosscompilation supported or documented by HaxeFlixel?

### DISCUSSION THREAD

**Discussion by MetalMaxMX** [HaxeFlixel/flixel#2892] [Q&A] [answered]
Hi! I've been checking the docs but I haven't been able to find any proper docs on how to export binaries from one OS to another entirely. Let's say I am on Linux and I want to export a build with Windows using MinGW. How would that be achieved? Are there any tools that facilitate such behavior? On my quest to find answers, I only got this SO exchange about it 
https://stackoverflow.com/questions/32516679/how-to-cross-compile-haxe-openfl-generated-source-code-using-mingw

There was also this post but I am not sure if it is still relevant to how Flixel works with 
https://community.openfl.org/t/compiling-windows-cpp-from-linux-using-mingw-its-possible/6503/2
And this PR https://github.com/HaxeFoundation/hxcpp/issues/282

But I am pretty sure you can't easily do a "lime test windows" from Linux without either a fork of Lime (that's not even archived it seems?) or the like, right? Just wanting to know about this specific scenario.

**Comment by Vortex2Oblivion** [comment, marked as answer]
There isn't really a great way to cross compile for flixel afaik. What I've been doing is setting up a GitHub workflow file to compile a project to different targets when I commit to a repository. That's what I would recommend, the only other option really is to get a friend to compile it for windows / macos / linux or do it on a separate machine or vm.

**Reply by MetalMaxMX** [reply]
Bit of a shame, but that's understandable. I suppose setting up toolchains surrounding MinGW is something out of the scope for this?

**Reply by Vortex2Oblivion** [reply]
If you're on Linux you might be able to build to windows by installing the windows version of haxe via wine. I could very much be wrong, though as I have never tried this and use windows myself.

**Reply by MetalMaxMX** [reply]
That's going to be difficult unless HaxeFlixel (and by extension OpenFL) supported building through MinGW, then it won't be really possible because Visual Studio and, more importantly, MSVC doesn't work on Wine. So you wouldn't be able to use the C++ target anyway from Linux

**Comment by MetalMaxMX** [comment]
I'd like to give an update. If we investigate what Lime is currently doing, it seems the road for cross-compilation on other platforms doesn't seem to be too far away, as there are open commits waiting to be merged. Albeit the discussion seems to be around if passing an -hl (HashLink) flag or a MinGW flag -mingw.

https://github.com/openfl/lime/pull/1610
https://github.com/openfl/lime/pull/1661
https://github.com/openfl/lime/pull/1660

If all of these are somehow merged or find a way to document and establish this functionality, then, is likely that HaxeFlixel would also now compile through a Linux or macOS host it seems. Awaiting to see how it goes.
